modules/core/include/opencv2/core/quaternion.inl.hpp

```diff
@@ -866,6 +866,766 @@ Quat<T> Quat<T>::spline(const Quat<T> &q0, const Quat<T> &q1, const Quat<T> &q2,
     return squad(vec[1], s1, s2, vec[2], t, assumeUnit, QUAT_ASSUME_NOT_UNIT);
 }
 
+template <typename T>
+Quat<T> Quat<T>::createFromYRot(const T theta)
+{
+    return Quat<T>{std::cos(theta), 0, std::sin(theta), 0};
+}
+
+template <typename T>
+Quat<T> Quat<T>::createFromXRot(const T theta){
+    return Quat<T>{std::cos(theta), std::sin(theta), 0, 0};
+}
+
+template <typename T>
+Quat<T> Quat<T>::createFromZRot(const T theta){
+    return Quat<T>{std::cos(theta), 0, 0, std::sin(theta)};
+}
+
+template <typename T>
+Quat<T> Quat<T>::createFromEulerAngles(const Vec<T, 3> &angles,EulerAnglesType eulerAnglesType) {
+    Quat<T> q;
+    switch (eulerAnglesType)
+    {
+    case INT_XYZ:
+    {
+        Quat<T> q1 =  Quat<T>::createFromXRot(angles(0) / 2.);
+        Quat<T> q2 =  Quat<T>::createFromYRot(angles(1) / 2.);
+        Quat<T> q3 =  Quat<T>::createFromZRot(angles(2) / 2.);
```

Comment:
> / 2.

`* 0.5f`

Reply:
OK! Thanks for your reviewing code for us.